vcd/resource_vcd_org_group.go

```diff
@@ -140,30 +159,30 @@ func resourceVcdOrgGroupUpdate(d *schema.ResourceData, meta interface{}) error {
 	if d.HasChange("description") {
 		group.Group.Description = d.Get("description").(string)
 	}
-
+	group.Group.UsersList = nil
```

Comment:
Ha, good catch! Actually I put this when I was debugging, indeed it was VCD complaining. But now I'm not sure if it was due to a wrong usage I eventually fixed or not, so I'd re-check this one and provide some conclusion.

Reply:
Ok, just ping once you are done with that and I can give it a manual run test before approving.

Reply:
Okay, I tested this line and I remembered. VCD responds with an error 400 because it thinks you're willing to change the users in the list:
`HTTP 400 Bad Request: Invalid content was found starting with element 'UsersList'. No child element is expected at this point`

Now, probably it's more correct to encapsulate that in the `group.Update()` operation for groups.

Reply:
I am good keeping it here. We most often try to leave such things for user instead of embedding something into built in functions (like `Update()` - the reason is - maybe this changes one day and then we have a bug report for SDK + that implies behavioral change.

Reply:
I am good keeping it here. We most often try to leave such things for user instead of embedding something into built in functions (like `Update()` - the reason is - maybe this changes one day and then we have a bug report for SDK + that implies behavioral change.

Reply:
Finally I've removed this `nil` assignment and improved the Go SDK here:
https://github.com/vmware/go-vcloud-director/pull/449
The reason is to avoid strange quirks using the SDK, as in most cases the assignment to `nil` would be required for the `Update()` to work.